jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/internal/HttpReceiverOverHTTP.java

```diff
@@ -550,7 +550,7 @@ public void badMessage(BadMessageException failure)
         {
             HttpResponse response = exchange.getResponse();
             response.status(failure.getCode()).reason(failure.getReason());
-            failAndClose(new HttpResponseException("HTTP protocol violation: bad response on " + getHttpConnection(), response, failure));
+            failAndClose(new HttpResponseException("HTTP protocol violation: bad response on " + getHttpConnection(), response, HttpException.asThrowable(failure)));
```

Comment:
Honestly, I would just cast: `_failure = (Throwable)failure;`.
I'd prefer to fail the cast, rather than having `_failure==null` in the unlikely case we mess up and make one non-exception class implement `HttpException`.